sdk/include/opentelemetry/sdk/metrics/metric_reader.h

```diff
@@ -5,6 +5,7 @@
 
 #include <atomic>
 #include <chrono>
+#include <memory>
```

Comment:
This line is no longer needed, now that the shared_from_this moved to the sub class.

Please note that the include-what-you-use build in CI now reports this:

```
[ 25%] Building CXX object sdk/src/metrics/CMakeFiles/opentelemetry_metrics.dir/metric_reader.cc.o
Warning: include-what-you-use reported diagnostics:

/home/runner/work/opentelemetry-cpp/opentelemetry-cpp/sdk/include/opentelemetry/sdk/metrics/metric_reader.h should add these lines:

/home/runner/work/opentelemetry-cpp/opentelemetry-cpp/sdk/include/opentelemetry/sdk/metrics/metric_reader.h should remove these lines:
- #include <memory>  // lines 8-8

The full include-list for /home/runner/work/opentelemetry-cpp/opentelemetry-cpp/sdk/include/opentelemetry/sdk/metrics/metric_reader.h:
#include <atomic>                                              // for atomic
#include <chrono>                                              // for micros...
#include "opentelemetry/nostd/function_ref.h"                  // for functi...
#include "opentelemetry/sdk/metrics/export/metric_producer.h"  // for Metric...
#include "opentelemetry/sdk/metrics/instruments.h"             // for Aggreg...
#include "opentelemetry/version.h"                             // for OPENTE...
```

Warnings are not enforced as failures (yet), which is why it passes.

Reply:
Thanks, some other IWYU problems are also fixed.